concepts/blocks/about.md

```diff
@@ -57,3 +57,104 @@ int main() {
 If the last `printf` statement is removed, the program will compile and print `0 2 4 6 8 10 ` when run.
 Some languages, such as Go and Rust, are explicit that single-line conditional and loop statements constitute a block.
 They require that such single-line statements still be enclosed in curly braces.
+
+A function acts somewhat like a block, but functions and blocks are not identical.
+In the following example, the code will not compile
+
+```c
+#include <stdio.h>
+
+int main() {
+
+  {
+      int i = 0;
+      begin:
+      printf("i == %d\n", i);
+      if (i == 0) {
+          i = 1;
+          goto begin;
+      }
+  }
+
+  {
+      int i = 1;
+      begin:
+      printf("i == %d\n", i);
+      if (i == 1) {
+          i = 0;
+          goto begin;
+      }
+  }
+
+}
+```
+
+The code above fails to compile with the following error
+
+```
+/tmp/gH2TVYnTSJ.c: In function 'main':
+/tmp/gH2TVYnTSJ.c:17:5: error: duplicate label 'begin'
+   17 |     begin:
+      |     ^~~~~
+/tmp/gH2TVYnTSJ.c:7:5: note: previous definition of 'begin' was here
+    7 |     begin:
+      |     ^~~~~
+```
+
+Although `i` can be defined in each block, `label` names must be unique in a function.
+That is because `labels` have function scope, meaning each label is seen by the function no matter where in the function it is defined.
+So, unlike variables, no matter where a `label` is defined in a function it must have a unique name within that same function.
+To make the code work requires changing the name of one of the `labels` so they are both unique, as in the following example
+
+```c
+#include <stdio.h>
+
+int main() {
+
+    {
+        int i = 0;
+        begin:
+        // prints
+        // i == 0
+        // i == 1
+        printf("i == %d\n", i);
+        if (i == 0) {
+            i = 1;
+            goto begin;
+        }
+    }
+
+    {
+        int i = 1;
+        start:
+        // prints
+        // i == 1
+        // i == 0    
+        printf("i == %d\n", i);
+        if (i == 1) {
+            i = 0;
+            goto start;
+        }
+    }
+
+}
+```
+
+Single-line statements can be "stacked" and still be within one block without the use of curly braces.
+In the example below, the `for` statement contains an `if` statement which in turn contains a `printf` statement.
+Although there are two total statements in the `for` block, it "sees" only its single-line `if` statement.
```

Comment:
`"sees"` is kind of how I think of it too... but I feel like there is a more correct way to phrase this.

Something like "Although there are two total statements in the `for` block, only its single-line `if` statement is evaluated as the body of the `for` loop.".
What do you think?

Reply:
Followed your recommendation and changed to this

```
Although there are two total statements in the `for` block, only its single-line `if` statement is evaluated as the body of the `for` loop.
The `if` statement's single-line `printf` statement in turn is evaluated as its body.
If nested blocks are each only a single line, none of the blocks need curly braces.
```